packages/block-library/src/site-title/edit.js

```diff
@@ -27,13 +38,28 @@ export default function SiteTitleEdit() {
 			>
 				{ __( 'Update' ) }
 			</Button>
-			<RichText
-				tagName="h1"
+			<PlainText
 				placeholder={ __( 'Site Title' ) }
 				value={ title }
 				onChange={ setTitle }
-				allowedFormats={ [] }
+				isStylable
+				onKeyDown={ preventNewlines }
```

Comment:
I think instead of `onKeyDown` here, we might want to align the API with how RichText works, I think it's `onSplit` or something like that cc @ellatrix

Reply:
The difference here is that `onSplit` is an actual `RichText` [method](https://github.com/WordPress/gutenberg/blob/5d0c03e39ecd69ee814bb7f938373bfee96a6928/packages/block-editor/src/components/rich-text/index.js#L194-L253) wrote for handling a block split, while in this case I'm just passing the native `onKeyDown` handler through to the `textarea` element.

Reply:
This is another reason why I think it would be better to do this through (Rich)Text. The API's already there, and they would be the same.